internal/config/config.go

```diff
@@ -323,6 +323,17 @@ func Load(cfg *Config) *viper.Viper {
 		slog.Warn("Mapping to configuration", slog.Any("error", err))
 	}
 
+	sec, err := secrets.NewAppSecrets(cfg.Application.MainSecret)
+	if err != nil {
+		panic(err)
+	}
+	cfg.Secrets = *sec
+
+	if err := Validate(cfg); err != nil {
+		reportError(err)
+		panic("Invalid configuration")
+	}
+
```

Comment:
Why did you move the code around and move `NewAppSecrets` before `Validate`? The `NewAppSecrets` function assumes that the configuration has been properly validated.

Reply:
Thanks

(1) I moved the code to `Load` method; (2) I changed the order because I observed the configuration `cfg.Secrets` field was updated; so thinking the validation should be made once that value was written.

I restore (2) to keep the order between `NewAppSecrets` and `Validate`

Reply:
The sub-structure is ignored by validation, because it is an untagged field:

```
        // Secrets is an untagged field and filled out on load
        Secrets secrets.AppSecrets `mapstructure:"-"`
```